### PR TITLE
Deployments with deprecated runtimes

### LINKED ISSUE #3552: Add notes for deprecated runtime
### Issue Type

Missing content for an existing feature

### Links to Affected Docs

- [ ] https://docs.astronomer.io/astro/migrate-mwaa#prerequisites
- [ ] https://docs.astronomer.io/astro/migrate-gcc#prerequisites
- [x] https://docs.astronomer.io/astro/create-deployment
- [x] https://docs.astronomer.io/astro/upgrade-runtime#upgrade-considerations

### The Issue or Context

We have a new functionality that allows users to run Deployments with deprecated runtimes after contacting CRE to enable the feature. 

- Add a central place for that information (Upgrade Runtimes page) 
- Add notes on other relevant pages to point to more detailed section

### Fix or Proposed Change

Referral notes need to include:

- Link to the *Create Deployment* section of the Platform API docs (https://docs.astronomer.io/astro/api/platform-api-reference#tag/Deployment)
- Link to main note about process in Upgrade Runtime section

Main section

- Includes Link to the *Create Deployment* section of the Platform API docs (https://docs.astronomer.io/astro/api/platform-api-reference#tag/Deployment)
- Describes behavior "won't be able to create a deployment for a version of runtime that is not supported"/ yanked
- Describes when to use this feature ie migrating from OSS airflow or Software/hybrid to hosted

### Additional Notes

- Restricted deprecated runtime https://github.com/astronomer/astro-runtime/pull/1379/files
- https://github.com/astronomer/astro/pull/19826

### Required Reviewers

frank ye

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #3552 

Relates to 
* https://github.com/astronomer/astro/pull/19826
* https://github.com/astronomer/astro-runtime/pull/1379/files

### CODE REVIEW

File: astro/create-deployment.md [hunk 1]
```diff
@@ -29,6 +29,15 @@ After you create a Deployment, you can deploy DAGs to it using the Astro CLI on
 
 Every Deployment is hosted on an Astro cluster with its own dedicated resources that you can [customize](deployment-resources.md) to fine-tune your resource usage. To restrict communication between Deployments, resources for each Deployment are isolated within a corresponding Kubernetes namespace. See [Deployment network isolation](data-protection.md#deployment-network-isolation).
 
+:::note
+
+If you're creating a Deployment to migrate to Astro Hosted from OSS Airflow, Astro Hybrid, or Software, and you need to use a deprecated version of the [Astro Runtime](runtime-version-lifecycle-policy.md), you can create Deployments with [deprecated versions of the Astro Runtime](upgrade-runtime.md#run-a-deprecated-astro-runtime).
```
Comment (frank-ye) [inline astro/create-deployment.md:34]: I don't think this applies to Software unfortunately.

Comment (frank-ye) [reply astro/create-deployment.md:34]: Unless someone on the software team has also worked on this issue, @paolaperaza do you mind confirming here please?

Comment (lzdanski) [reply astro/create-deployment.md:34]: I think you're right! I misread Paola's first message. Making change.

Comment (lzdanski) [inline astro/create-deployment.md:34]:
```suggestion
If you're creating a Deployment to migrate to Astro Hosted from OSS Airflow or Astro Hybrid, and you need to use a deprecated version of the [Astro Runtime](runtime-version-lifecycle-policy.md), you can create Deployments with [deprecated versions of the Astro Runtime](upgrade-runtime.md#run-a-deprecated-astro-runtime).
```

File: astro/upgrade-runtime.md [hunk 2]
```diff
@@ -120,6 +120,24 @@ Consider the following when you upgrade Astro Runtime:
 
 To stay up to date on the latest versions of Astro Runtime, see [Astro Runtime release notes](runtime-release-notes.md). For more information on Astro Runtime versioning and support, see [Astro Runtime versioning and lifecycle policy](runtime-version-lifecycle-policy.md). For a full collection of Astro Runtime Docker images, go to the [Astro Runtime repository on Quay.io](https://quay.io/repository/astronomer/astro-runtime?tab=tags).
 
+### Run a deprecated Astro Runtime
+
+<HostedBadge/>
+
+If you're migrating to Astro from OSS Airflow, Software, or Astro Hybrid, where your Deployments run using an older version of Airflow or a deprecated version of the Astro Runtime, you can now use stepwise migration to complete your migration in stages. First, you can create Deployments with a deprecated version of the Astro Runtime using the Astro API to move to an Astro Hosted environment. Then, you can upgrade your code to the most up-to-date version of the Astro Runtime.
```
Comment (lzdanski) [inline astro/upgrade-runtime.md:127]:
```suggestion
If you're migrating to Astro from OSS Airflow or Astro Hybrid, where your Deployments run using an older version of Airflow or a deprecated version of the Astro Runtime, you can now use stepwise migration to complete your migration in stages. First, you can create Deployments with a deprecated version of the Astro Runtime using the Astro API to move to an Astro Hosted environment. Then, you can upgrade your code to the most up-to-date version of the Astro Runtime.
```

File: astro/upgrade-runtime.md [hunk 3]
```diff
@@ -120,6 +120,24 @@ Consider the following when you upgrade Astro Runtime:
...
 
+### Run a deprecated Astro Runtime
+
+<HostedBadge/>
+
+If you're migrating to Astro from OSS Airflow or Astro Hybrid, where your Deployments run using an older version of Airflow or a deprecated version of the Astro Runtime, you can now use stepwise migration to complete your migration in stages. First, you can create Deployments with a deprecated version of the Astro Runtime using the Astro API to move to an Astro Hosted environment. Then, you can upgrade your code to the most up-to-date version of the Astro Runtime.
+
+When you create a Deployment with a deprecated Astro Runtime version, Astronomer Support still follows the [Astro Runtime maintenance policy](runtime-version-lifecycle-policy.md#astro-runtime-maintenance-policy), and might advise you to upgrade your runtime to the latest version of Astro to resolve performance issues or bugs.
+
+:::note
```
Comment (jwitz) [inline astro/upgrade-runtime.md:131]:
```suggestion
:::info
```

File: astro/upgrade-runtime.md [hunk 4]
```diff
@@ -120,6 +120,24 @@ Consider the following when you upgrade Astro Runtime:
...
+
+There are some versions of deprecated Astro Runtime that included undesirable behavior or bugs in the version of either Airflow or Astro Runtime. If you try to use one of these versions when creating a Deployment with a deprecated Runtime, the Astro API returns a failure. See the [Version upgrade considerations](#version-upgrade-considerations) for additional information when choosing a Runtime version to use.
```
Comment (jwitz) [inline astro/upgrade-runtime.md:133]:
```suggestion
When you choose a deprecated version of Astro Runtime, Astronomer recommends always choosing the latest patch version that includes your desired Airflow version. Some earlier patch versions of Astro Runtime include unexpected behavior and can't be used even as a deprecated version. For more information about which versions are affected, see the [Version upgrade considerations](#version-upgrade-considerations).
```

File: astro/upgrade-runtime.md [hunk 5]
```diff
@@ -120,6 +120,24 @@ Consider the following when you upgrade Astro Runtime:
...
+
+:::
+
+1. Contact [Astronomer Suppor](astro-support.md) to allow your Deployments to run deprecated versions of the Astro Runtime.
```
Comment (jwitz) [inline astro/upgrade-runtime.md:137]:
```suggestion
1. Contact [Astronomer support](astro-support.md) and request the ability to run deprecated versions of Astro Runtime.
```

File: astro/upgrade-runtime.md [hunk 6]
```diff
@@ -120,6 +120,24 @@ Consider the following when you upgrade Astro Runtime:
...
+2. Create either a Dedicated or Standard Deployment using the [Astro API](https://docs.astronomer.io/api/platform-api-reference/deployment/create-deployment), and define the `astroRuntimeVersions` using the deprecated Runtime version you want to use.
```
Comment (jwitz) [inline astro/upgrade-runtime.md:138]:
```suggestion
2. Create a Deployment using the [Astro API](https://docs.astronomer.io/api/platform-api-reference/deployment/create-deployment). In your request, specify your deprecated Astro Runtime version using the `astroRuntimeVersions` property.
```

File: astro/upgrade-runtime.md [hunk 7]
```diff
@@ -120,6 +120,24 @@ Consider the following when you upgrade Astro Runtime:
...
+3. (Optional) After you create the Deployment, you can manage settings and other features using the [Astro UI](deployment-settings.md).
```
Comment (jwitz) [inline astro/upgrade-runtime.md:139]:
```suggestion

After you create the Deployment, you can manage the Deployment using any interface, including the Astro CLI and [Astro UI](deployment-settings.md).
```

**Review by jwitz — APPROVED** [review]
LGTM, just tried to simplify some of the migration language and some of the language about yanked Runtime versions

File: astro/create-deployment.md [hunk 8]
```diff
@@ -29,6 +29,15 @@ After you create a Deployment, you can deploy DAGs to it using the Astro CLI on
 
 Every Deployment is hosted on an Astro cluster with its own dedicated resources that you can [customize](deployment-resources.md) to fine-tune your resource usage. To restrict communication between Deployments, resources for each Deployment are isolated within a corresponding Kubernetes namespace. See [Deployment network isolation](data-protection.md#deployment-network-isolation).
 
+:::note
+
+If you're creating a Deployment to migrate to Astro Hosted from OSS Airflow or Astro Hybrid, and you need to use a deprecated version of the [Astro Runtime](runtime-version-lifecycle-policy.md), you can create Deployments with [deprecated versions of the Astro Runtime](upgrade-runtime.md#run-a-deprecated-astro-runtime).
```
Comment (lzdanski) [inline astro/create-deployment.md:34]:
```suggestion
If you're creating a Deployment to migrate to Astro Hosted from OSS Airflow or Astro Hybrid, and you need to use a deprecated version of the [Astro Runtime](runtime-version-lifecycle-policy.md), you can create Deployments with [deprecated versions of the Astro Runtime](upgrade-runtime.md#run-a-deprecated-astro-runtime-version).
```

**Review by frank-ye — APPROVED** [review]

File: astro/create-deployment.md [hunk 9]
```diff
@@ -29,6 +29,14 @@ After you create a Deployment, you can deploy DAGs to it using the Astro CLI on
 
 Every Deployment is hosted on an Astro cluster with its own dedicated resources that you can [customize](deployment-resources.md) to fine-tune your resource usage. To restrict communication between Deployments, resources for each Deployment are isolated within a corresponding Kubernetes namespace. See [Deployment network isolation](data-protection.md#deployment-network-isolation).
 
+:::info
+
+If you're creating a Deployment to migrate to Astro Hosted from OSS Airflow or Astro Hybrid, and you need to use a deprecated version of the [Astro Runtime](runtime-version-lifecycle-policy.md), you can create Deployments with [deprecated versions of the Astro Runtime](upgrade-runtime.md#run-a-deprecated-astro-runtime-version).
```
Comment (paolaperaza) [inline astro/create-deployment.md:34]:
```suggestion
If you're migrating to Astro from OSS Airflow or another Astronomer product and you currently use an older version of Airflow, you can still create Deployments with the corresponding version of Astro Runtime even if it is deprecated according to the [Astro Runtime maintenance policy](runtime-version-lifecycle-policy.md#astro-runtime-maintenance-policy). This allows you to migrate your DAGs to Astro without needing to make any code changes and immediately upgrade to a new version of Airflow. Note that once you migrate your DAGs, Astronomer recommends upgrading to a supported version of Astro Runtime as soon as you can.

See [Run a deprecated Astro Runtime version](upgrade-runtime.md#run-a-deprecated-astro-runtime-version). 
```

Comment (paolaperaza) [reply astro/create-deployment.md:34]: @lzdanski I happened upon this PR as I was prepping for release preview and changed up wording! Sorry, feel free to take or leave

**Review by paolaperaza — APPROVED** [review]

File: docs-partials/prereqs.mdx [hunk 10]
```diff
@@ -17,6 +17,14 @@ On the cloud service from which you're migrating, ensure that you have:
...
 - Permission to create new repositories on your source control tool.
 - (Optional) Access to your secrets backend.
-- (Optional) Permission to create new CI/CD pipelines. 
+- (Optional) Permission to create new CI/CD pipelines.
 
 All source Airflow environments on 1.x need to be upgraded to at least Airflow 2.0 before you can migrate them. Astronomer professional services can help you with the upgrade process.
+
+:::info
+
+If you're migrating to Astro from OSS Airflow or another Astronomer product, and you currently use an older version of Airflow, you can still create Deployments with the corresponding version of Astro Runtime even if it is deprecated according to the [Astro Runtime maintenance policy](runtime-version-lifecycle-policy.md#astro-runtime-maintenance-policy). This allows you to migrate your DAGs to Astro without needing to make any code changes and then immediately upgrade to a new version of Airflow. Note that after you migrate your DAGs, Astronomer recommends upgrading to a supported version of Astro Runtime as soon as you can.
```
Comment (lzdanski) [inline docs-partials/prereqs.mdx:26]:
```suggestion
If you're migrating to Astro from OSS Airflow or another Astronomer product, and you currently use an older version of Airflow, you can still create Deployments with the corresponding version of Astro Runtime even if it is deprecated according to the [Astro Runtime maintenance policy](https://docs.astronomer.io/astro/runtime-version-lifecycle-policy#astro-runtime-maintenance-policy). This allows you to migrate your DAGs to Astro without needing to make any code changes and then immediately upgrade to a new version of Airflow. Note that after you migrate your DAGs, Astronomer recommends upgrading to a supported version of Astro Runtime as soon as you can.
```

**Pull request merged**